backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/log/ModularContractLogDAO.java

```diff
@@ -105,19 +109,19 @@ public ModularContractLogEntryId create(@NonNull final LogEntryCreateRequest req
 			log.setC_Invoice_Candidate_ID(invoiceCandidateId.getRepoId());
 		}
 
-		final Money amount = request.getAmount();
-		if (amount != null)
-		{
-			log.setAmount(amount.toBigDecimal());
-			log.setC_Currency_ID(amount.getCurrencyId().getRepoId());
-		}
+		Optional.ofNullable(request.getAmount())
+				.ifPresentOrElse(amount -> {
+									 log.setAmount(amount.toBigDecimal());
+									 log.setC_Currency_ID(amount.getCurrencyId().getRepoId());
+								 },
+								 () -> log.setAmount(null));
 
-		final Quantity quantity = request.getQuantity();
-		if (quantity != null)
-		{
-			log.setQty(quantity.toBigDecimal());
-			log.setC_UOM_ID(quantity.getUomId().getRepoId());
-		}
+		Optional.ofNullable(request.getQuantity())
```

Comment:
hmmm, if you would like to go the Optionals way, 
IMHO request.getQuantity() shall return Optional.